Tests/OptimizelyTests-DataModel/AttributeValueTests.swift

```diff
@@ -346,11 +346,11 @@ extension AttributeValueTests {
         let attr = AttributeValue.others
         var tmpError: Error?
         do {
-            _ = try attr.isExactMatch(with: invalidValue, condition: conditionString, name: attributeKey)
+            _ = try attr.isExactMatch(with: invalidValue)
         } catch {
             tmpError = error
         }
-        XCTAssertEqual("[Optimizely][Error] " + tmpError!.localizedDescription, OptimizelyError.evaluateAttributeInvalidCondition(conditionString).localizedDescription)
+        XCTAssertNotNil(tmpError)
```

Comment:
Can we change these non-nil checking to specific error type checking?

`
var tmpError: OptimizelyError?
...
        } catch {
            tmpError = error as? OptimizelyError
        }
... 
if case . evaluateAttributeInvalidCondition = tmpError {
    XCTAssert(true)
} else {
   XCTAssert(false)
}
`

Reply:
I removed specific error type checking here since we are doing the same in the new decisionService tests.

Reply:
I understand they're redundant. but it's better to have type checking here. Without them, these tests look incomplete.
Recovering your old XCTAssertEqual will also work.